lua/entities/kleiner/init.lua

```diff
@@ -0,0 +1,948 @@
+            RELEVANT_KLEINER = self -- see ENT.PathGen for explanation
+            KLPATHGEN_ITERS = 0
+            KLPATHGEN_ITERS_BUDGET = KleinerPathingIterationLimit()
+            local success = path:Compute(self, self:GetTarget():GetPos(), self.PathGen)
+            if (not success) then return "failed" end
+        end
+
+        if (not path:IsValid()) then return "failed" end
+        local shouldpath = self:WhilePathing(path)
+
+        if (shouldpath) then
+            path:Update(self)
+        end
+
+        if (options.draw) then
+            path:Draw()
+        end
+
+        if (self.loco:IsStuck()) then
+            self:HandleStuck()
+
+            return "stuck"
+        end
+
+        coroutine.yield()
+    end
+
+    return "ok"
+end
+
+function ENT:OnContact(ent)
+
+end
+
+local function IsDoor(door)
+
+    local doorClass = door:GetClass()
+	if ( doorClass == "func_door" or doorClass == "func_door_rotating" or  doorClass == "prop_door_rotating" ) then
+        return true
+    end
+    return false
+end
+local function DoorIsOpen( door )
+	
+	local doorClass = door:GetClass()
+
+	if ( doorClass == "func_door" or doorClass == "func_door_rotating" ) then
+
+		return door:GetInternalVariable( "m_toggle_state" ) == 0
+
+	elseif ( doorClass == "prop_door_rotating" ) then
+
+		return door:GetInternalVariable( "m_eDoorState" ) ~= 0
+
+	else
+
+		return false
+
+	end
+
+end
+
+--this function runs during the path movement. returning false will interrupt path movement, in case other actions are needed.
+function ENT:WhilePathing(path)
+    if (path == nil or not path:IsValid()) then return true end
+    if (self.loco == nil) then return true end
+    if (not self:OnGround() and self:GetVelocity().z < -50) then return false end --attempting to move while falling seems to pause falling
+    local seg1, index = self:GetCurrentPathPoint() --this returns the first path segment we're closest to.
+    local seg2 = path:GetAllSegments()[index + 1]
+    local seg3 = path:GetAllSegments()[index + 2]
+    local dir = self:GetForward()
+    local tseg1 = seg1
+    local tseg2 = seg3 or seg2
+    
+
+
+    if(tseg1 != nil and tseg2 != nil)then
+        dir = (tseg2.pos - tseg1.pos):GetNormalized()
+
+    local tr = {}
+    tr.start = self:GetPos() + Vector(0,0,30)
+    tr.endpos = tr.start + dir*100
+    tr.filter = self
+    tr.mask = MASK_PLAYERSOLID
+        if(self:GetVelocity():Length() < 1)then
+            local trace = util.TraceLine(tr)   
+            if(trace.Hit and IsDoor(trace.Entity))then
+                
+                if( !DoorIsOpen( trace.Entity ))then
+                    print("     ATTEMPTING TO OPEN A DOOR!!!")
+                    trace.Entity:Use(self,self,SIMPLE_USE,1)
+                    trace.Entity.KleinerInteracting = true
+                    self.loco:SetVelocity(dir*-500)
+                end
+                timer.Create(trace.Entity:EntIndex().."kleinerdoor",2,1,function()
```

Comment:
KleinerInteracting isn't used anywhere.

I'm sorry but this kind of thing is just sloppy. You need to be cleaning up your code. This is a very common theme with things you submit - unused variables, functions, logic etc, that was never cleaned up; extraneous features that were never finished or got removed but only partially, and so on.

**This is by far the biggest problem with your development habits.** Unused code should ALWAYS be deleted or at least commented out. I haven't been complaining too much about it because I can't hold you to "professional" standards but this is just getting annoying to me.

If you look at any "real" repository you will virtually never see this kind of unused/unfinished/partially-removed logic anywhere and for good reason. When you leave this stuff in it just creates confusion for the reader (me) who will inevitably have to modify or debug this in the future.

Reply:
I will make an effort to do cleanup when i do the other fixes later today. 
To be honest i kind of just assumed i had already done clean up on it months ago when i stopped working on it, which is a pretty stupid mistake on my part.

Reply:
This isn't just with this PR, it's pretty much everything you submit, and i've been putting up with it for a while but it is becoming apparent that this is just habitual for you and i'm telling you, you can't get away with this in the real world - ever.

Whenever I make a PR at my "real" job I always carefully read over every line of it and fix every little thing (even things like the grammar in comments) before submitting it because if I don't, the reviewer will see it anyway and make me fix it. The point of reviews isn't just to tell you to fix the specific PR you're submitting, it's to give you more direction on how to write the code correctly in the future so your reviewer's time isn't wasted on every single one of your PRs pointing out every single thing you should have fixed before submission.

Reply:
You're right. This is experience I don't have. I'm sorry this has come at the cost of your time, I am doing my best to get this under control